src/lib/mdns/minimal/ResponseSender.cpp

```diff
@@ -0,0 +1,159 @@
+    {                                                                                                                              \
+        if (err)                                                                                                                   \
+        {                                                                                                                          \
+            return;                                                                                                                \
+        }                                                                                                                          \
+    } while (false)
+
+namespace mdns {
+namespace Minimal {
+
+namespace {
+
+constexpr uint16_t kMdnsStandardPort = 5353;
+constexpr uint16_t kPacketSizeBytes  = 512;
+
+} // namespace
+
+CHIP_ERROR ResponseSender::Respond(const QueryData & query, const chip::Inet::IPPacketInfo * querySource)
+{
+    mSendError = CHIP_NO_ERROR;
+
+    mCurrentSource = querySource;
+    mSendUnicast   = query.GetUnicastAnswer();
+
+    mResponder->ResetAdditionals();
+
+    mCurrentResourceType = ResourceType::kAnswer; // direct answer
+    QueryReplyFilter filter(query);
+    for (auto it = mResponder->begin(); it != mResponder->end(); it++)
+    {
+        Responder * responder = it->responder;
+
+        if (!filter.SendAnswer(responder->GetQType(), responder->GetQClass(), responder->GetQName()))
+        {
+            continue;
+        }
+
+        responder->AddAllResponses(querySource, this);
+        ReturnErrorOnFailure(mSendError);
+
+        mResponder->MarkAdditionalReplyesFor(it);
+    }
+
+    mCurrentResourceType = ResourceType::kAdditional; // Additional parts
+    filter.SetIgnoreNameMatch(true);
+    for (auto it = mResponder->additional_begin(); it != mResponder->additional_end(); it++)
+    {
+        Responder * responder = it->responder;
+
+        if (!filter.SendAnswer(responder->GetQType(), responder->GetQClass(), responder->GetQName()))
+        {
+            continue;
+        }
+
+        it->responder->AddAllResponses(querySource, this);
+        ReturnErrorOnFailure(mSendError);
+    }
+
+    return FlushReply();
+}
+
+CHIP_ERROR ResponseSender::FlushReply()
+{
+    ReturnErrorCodeIf(mCurrentPacket.IsNull(), CHIP_NO_ERROR); // nothing to flush
+
+    if (mResponseBuilder.HasResponseRecords())
+    {
+
+        if (!mSendUnicast && (mCurrentSource->SrcPort == kMdnsStandardPort))
+        {
+            ChipLogProgress(Discovery, "Broadcasting mDns reply");
+            ReturnErrorOnFailure(
+                mServer->BroadcastSend(mCurrentPacket.Release_ForNow(), kMdnsStandardPort, mCurrentSource->Interface));
+        }
+        else
+        {
+            ChipLogProgress(Discovery, "Directly sending mDns reply to peer on port %d", mCurrentSource->SrcPort);
+            ReturnErrorOnFailure(mServer->DirectSend(mCurrentPacket.Release_ForNow(), mCurrentSource->SrcAddress,
```

Comment:
Could you send me the link for the spec in this?

I am surprised since it is obvious when a packet is received that it is not multicast, so I never thought I would need to set anything. It is possible, but more than one-line change so my instinct would be to bin this into 'not needed by the minMdns code'.

Reply:
Actually, unicast packets can be sent to the 5353 port IIRC. I may have misread. See https://tools.ietf.org/html/rfc6762#section-5.4